### ISSUE TITLE
coredump detector for pods

### ISSUE THREAD

**Issue by CaoShuFeng** [kubernetes/node-problem-detector#127]
I commit this issue to ask whether "coredump detector" is in the scope of this add-on.
I wrote a proposal about coredump in kubernetes cluster:
https://github.com/kubernetes/community/pull/658

But smarterclayton said we should not add a new api for coredump feature, we only need a privileged pod running on the node to capture this and report it via a different channel.

So, I am planning to add a coredump detector to each node, and upload coredump files to a separate storage backend. So that users can download coredump files from the storage backend.
This is the architecture:
![1](https://user-images.githubusercontent.com/20855209/28048966-d25dbfc2-6626-11e7-92eb-559e8ab22c59.png)

Is this the right place for the "coredump detector"?

**Comment by dchen1107** [comment]
I put my comment at the original proposal before I noticed this request. I copied the comment here:

Node-problem-detector (NPD) is designed to detect hardware, kernel, filesystem, etc. issues, which requires the remedy / repair actions from the cluster admin or auto management system. It is not designed for debugging application issues. Also NPD reports those problems to kube-apiserver, which defeats the purpose to report the applications' core dumps via a different channel commented by @smarterclayton at kubernetes/community#658
